packages/fxa-settings/src/pages/InlineTotpSetup/index.test.tsx

```diff
@@ -0,0 +1,157 @@
+    render(<InlineTotpSetup code={MOCK_CODE} />);
+    // const ftlMsgMock = screen.getAllByTestId('ftlmsg-mock')[1];
+    // testL10n(ftlMsgMock, bundle, {
+    //   email: exampleEmail,
+    // });
+    const continueButton = screen.getByRole('button', { name: 'Continue' });
+    fireEvent.click(continueButton);
+    await waitFor(() =>
+      expect(
+        screen.getByAltText(
+          `Use the code ${MOCK_CODE} to set up two-step authentication in supported applications.`
+        )
+      ).toBeInTheDocument()
+    );
+  });
+
+  it('toggles from QR code to manual secret code view when user clicks "Can\'t scan code"', async () => {
+    render(<InlineTotpSetup code={MOCK_CODE} />);
+    // const ftlMsgMock = screen.getAllByTestId('ftlmsg-mock')[1];
+    // testL10n(ftlMsgMock, bundle, {
+    //   email: exampleEmail,
+    // });
+    const continueButton = screen.getByRole('button', { name: 'Continue' });
+    fireEvent.click(continueButton);
+    await waitFor(() =>
+      expect(
+        screen.getByAltText(
+          `Use the code ${MOCK_CODE} to set up two-step authentication in supported applications.`
+        )
+      ).toBeInTheDocument()
+    );
+    const changeToManualModeButton = screen.getByRole('button', {
+      name: 'Can’t scan code?',
+    });
+    fireEvent.click(changeToManualModeButton);
+    await waitFor(() =>
+      expect(
+        screen.getByRole('button', { name: 'Scan QR code instead?' })
+      ).toBeInTheDocument()
+    );
+  });
+
+  it('toggles from secret code to QR code view when user clicks "Scan QR code instead?', async () => {
+    render(<InlineTotpSetup code={MOCK_CODE} />);
+    // const ftlMsgMock = screen.getAllByTestId('ftlmsg-mock')[1];
+    // testL10n(ftlMsgMock, bundle, {
+    //   email: exampleEmail,
+    // });
+    const continueButton = screen.getByRole('button', { name: 'Continue' });
+    fireEvent.click(continueButton);
+    await waitFor(() =>
+      expect(
+        screen.getByAltText(
+          `Use the code ${MOCK_CODE} to set up two-step authentication in supported applications.`
+        )
+      ).toBeInTheDocument()
+    );
+    const changeToManualModeButton = screen.getByRole('button', {
+      name: 'Can’t scan code?',
+    });
+    fireEvent.click(changeToManualModeButton);
+    await waitFor(() =>
+      expect(
+        screen.getByRole('button', { name: 'Scan QR code instead?' })
+      ).toBeInTheDocument()
+    );
+    fireEvent.click(
+      screen.getByRole('button', { name: 'Scan QR code instead?' })
+    );
+    await waitFor(() =>
+      expect(
+        screen.getByAltText(
+          `Use the code ${MOCK_CODE} to set up two-step authentication in supported applications.`
+        )
+      ).toBeInTheDocument()
+    );
+  });
+
+  it('emits the expected metrics on render', () => {
+    render(<InlineTotpSetup code={MOCK_CODE} />);
+    expect(logPageViewEvent).toHaveBeenCalledWith('inline-totp-setup', {
+      entrypoint_variation: 'react',
```

Comment:
FWIW I'm tired of seeing `{ entrypoint_variation: 'react', }` everywhere and I'm going to see about setting it to a const in my cleanup ticket.